src/server.rs

```diff
@@ -412,8 +414,14 @@ pub fn start_server(config: &Config, port: u16) -> Result<()> {
         }
         Err(e) => {
             error!("failed to start server: {}", e);
-            let reason = e.to_string();
-            notify_server_startup(&notify, ServerStartup::Err { reason })?;
+            if let Some(io_err) = e.downcast_ref::<io::Error>() {
+                if let io::ErrorKind::AddrInUse = io_err.kind() {
```

Comment:
This doesn't need to be `if let`, it could just be `if io_err.kind() == io::ErrorKind::AddrInUse {`. But furthermore, I believe you could write this as:
```rust
match e.downcast_ref::<io::Error>() {
  Some(io_err) if io_err.kind() == io::ErrorKind::AddrInUse => notify_server_startup(&notify, ServerStartup::AddrInUse)?;
  _ => {
    let reason = e.to_string();
    notify_server_startup(&notify, ServerStartup::Err { reason })?;
  }
}
```

Reply:
thanks